src/sync/update_tx.js

```diff
@@ -1,157 +1,214 @@
+async function executeFollowUpTx(tx, db) {
   const Transactions = db.Transactions;
-  if (approveTx.type === 'APPROVESETRESULT') {
-    let setResultTxid;
-    try {
-      const resp = await fetch('http://localhost:5555/set-result', {
-        method: 'POST',
-        body: JSON.stringify({
-          contractAddress: approveTx.entityId,
-          resultIndex: approveTx.optionIdx,
-          senderAddress: approveTx.senderAddress,
-        }),
-        headers: { 'Content-Type': 'application/json' },
-      }).then(res => res.json());
-      setResultTxid = resp.result.txid;
-    } catch (err) {
-      logger.error(`Error call /set-result: ${err.message}`);
-      throw err;
+  let txid;
+  switch (tx.type) {
+    // Approve was reset to 0. Sending approve for consensusThreshold.
+    case 'RESETAPPROVESETRESULT': {
+      try {
+        const approveTx = await bodhiToken.approve({
+          spender: tx.topicAddress,
+          value: tx.amount,
+          senderAddress: tx.senderAddress,
+        });
+        txid = approveTx.txid;
+      } catch (err) {
+        logger.error(`Error calling BodhiToken.approve: ${err.message}`);
+        throw err;
+      }
+
+      await DBHelper.insertTransaction(Transactions, {
+        _id: txid,
+        txid,
+        version: tx.version,
+        type: 'APPROVESETRESULT',
+        status: 'PENDING',
+        senderAddress: tx.senderAddress,
+        topicAddress: tx.topicAddress,
+        oracleAddress: tx.oracleAddress,
+        optionIdx: tx.optionIdx,
+        token: 'BOT',
+        amount: tx.amount,
+        createdTime: Date.now().toString(),
+      });
+      break;
     }
 
-    const tx = {
-      _id: setResultTxid,
-      version: approveTx.version,
-      type: 'SETRESULT',
-      txStatus: 'PENDING',
-      senderAddress: approveTx.senderAddress,
-      entityId: approveTx.entityId,
-      optionIdx: approveTx.optionIdx,
-      token: 'BOT',
-      amount: approveTx.amount,
-      createdTime: Date.now().toString(),
-    };
+    // Approve was accepted. Sending setResult.
+    case 'APPROVESETRESULT': {
+      try {
+        const setResultTx = await centralizedOracle.setResult({
+          contractAddress: tx.oracleAddress,
+          resultIndex: tx.optionIdx,
+          senderAddress: tx.senderAddress,
+        });
+        txid = setResultTx.txid;
+      } catch (err) {
+        logger.error(`Error calling CentralizedOracle.setResult: ${err.message}`);
+        throw err;
+      }
 
-    try {
-      await Transactions.insert(tx);
-    } catch (err) {
-      logger.error(`Error insert Transactions: ${err.message}`);
-      throw err;
+      await DBHelper.insertTransaction(Transactions, {
+        _id: txid,
+        txid,
+        version: tx.version,
+        type: 'SETRESULT',
+        status: 'PENDING',
+        senderAddress: tx.senderAddress,
+        oracleAddress: tx.oracleAddress,
+        optionIdx: tx.optionIdx,
+        token: 'BOT',
+        amount: tx.amount,
```

Comment:
convert amount from `decimalstring` to `hexstring`

Reply:
will leave as decimal string as discussed.

Reply:
same note as above